File: problem-016/main.cpp
#include <iostream>
#include <vector>
#include <numeric>


using namespace std;

void multiply(std::vector<int>& num, int factor)
{
	int rem = 0;

	for (int i=0; i<num.size(); i++)
	{
		int x = num[i] * factor + rem;
		num[i] = x % 10;
		rem = x / 10;
	}

	if (rem > 0)
	{
		num.push_back(rem);
	}
}



int main()
{
	std::vector<int> num (1, 1);
	int exponent = 1000;
	int base = 2;

	for (int i=0; i<exponent; i++)
	{
		multiply(num, base);
	}

	int result = accumulate(num.begin(), num.end(), 0);

	cout << result << endl;
}

File: problem-016/main.py

def manual_multiply(num: list, factor: int):
    rem = 0
    for i in range(len(num)):
        x = num[i] * factor + rem
        num[i] = x % 10
        rem = x // 10

    if rem > 0:
        num.append(rem)

    return num


def power(base, exponent):
    num = [1]
    for i in range(exponent):
        num = manual_multiply(num, base)
    return num


def listsum(arr):
    s = 0
    for x in arr:
        s += x
    return s


base = 2
exponent = 1000
result = listsum(power(base, exponent))

print(result)
        




